OpenImis.ModulesV2/PolicyModule/Repositories/PolicyRenewalRepository.cs

```diff
@@ -0,0 +1,268 @@
+        {
+            List<GetPolicyRenewalModel> response = new List<GetPolicyRenewalModel>();
+
+            try
+            {
+                using (var imisContext = new ImisDB())
+                {
+                    var officerCodeParameter = new SqlParameter("@OfficerCode", SqlDbType.NVarChar, 8);
+                    officerCodeParameter.Value = officerCode;
+
+                    var sql = "exec uspGetPolicyRenewals @OfficerCode";
+
+                    DbConnection connection = imisContext.Database.GetDbConnection();
+
+                    using (DbCommand cmd = connection.CreateCommand())
+                    {
+                        cmd.CommandText = sql;
+
+                        cmd.Parameters.AddRange(new[] { officerCodeParameter });
+
+                        if (connection.State.Equals(ConnectionState.Closed)) connection.Open();
+
+                        using (var reader = cmd.ExecuteReader())
+                        {
+                            do
+                            {
+                                while (reader.Read())
+                                {
+                                    response.Add(new GetPolicyRenewalModel()
+                                    {
+                                        RenewalId = int.Parse(reader["RenewalId"].ToString()),
+                                        PolicyId = int.Parse(reader["PolicyId"].ToString()),
+                                        OfficerId = int.Parse(reader["OfficerId"].ToString()),
+                                        OfficerCode = reader["OfficerCode"].ToString(),
+                                        CHFID = reader["CHFID"].ToString(),
+                                        LastName = reader["LastName"].ToString(),
+                                        OtherNames = reader["OtherNames"].ToString(),
+                                        ProductCode = reader["ProductCode"].ToString(),
+                                        ProductName = reader["ProductName"].ToString(),
+                                        VillageName = reader["VillageName"].ToString(),
+                                        RenewalPromptDate = reader["RenewalPromptDate"].ToString(),
+                                        Phone = reader["Phone"].ToString(),
+                                        RenewalUUID = GetRenewalUUIDById(int.Parse(reader["RenewalId"].ToString()))
+                                    });
+                                }
+                            } while (reader.NextResult());
+                        }
+                    }
+                }
+
+                return response;
+            }
+            catch (SqlException e)
+            {
+                throw e;
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        public int Post(PolicyRenewalModel policy)
+        {
+            int RV = 2;
```

Comment:
Add a TODO comment to take the response codes from a const list.

Reply:
Done